dev/domain-controller/new-agent-vm/Install-PCoIPAgent.ps1

```diff
@@ -1,41 +1,246 @@
+                $destFile = "C:\WindowsAzure\PCoIPAgentInstaller\" + $installerFileName
+                
+				Write-Verbose "Downloading PCoIP Agent"
+                Invoke-WebRequest $pcoipAgentInstallerUrl -OutFile $destFile
 
                 #install the agent
-                & "$dest\$installerFileName" /S
+				Write-Verbose "Installing PCoIP Agent"
+                $ret = Start-Process -FilePath $destFile -ArgumentList "/S /nopostreboot" -PassThru -Wait
+
+				# Check installer return code
+				if ($ret.ExitCode -ne 0) {
+					#exit code 1641 means requiring reboot machine after intallation is done, other non zere exit code means installation has some error
+					if ($ret.ExitCode -eq $EXIT_CODE_REBOOT) {
+						Write-Verbose "Request reboot machine after Installing pcoip agent."
+						# Setting the global:DSCMachineStatus = 1 tells DSC that a reboot is required
+						$global:DSCMachineStatus = 1
+					} else {
+						$errMsg = "Failed to install PCoIP Agent. Exit Code: " + $ret.ExitCode
+						Write-Verbose $errMsg
+						throw $errMsg
+					}
+				}
+				
+	            Write-Verbose "Finished PCoIP Agent Installation"
+            }
+        }
+
+        Script Register
+        {
+            DependsOn  = @("[Script]Install_PCoIPAgent")
+
+            GetScript  = { return 'registration'}
+            
+            TestScript = { 
+                cd "C:\Program Files (x86)\Teradici\PCoIP Agent"
+ 	            $ret = & .\pcoip-validate-license.ps1
+
+				# the powershell variable $? to indicate the last executing command status
+				return $?
+            }
+
+            SetScript  = {
+                #register code is stored at the password property of PSCredential object
+                $registrationCode = ($using:registrationCodeCredential).GetNetworkCredential().password
+                if ($registrationCode) {
+					# Insert a delay before activating license
+	                cd "C:\Program Files (x86)\Teradici\PCoIP Agent"
+
+					Write-Verbose "Activating License Code"               
+ 	                $ret = & .\pcoip-register-host.ps1 -RegistrationCode $registrationCode
+					$isExeSucc = $?
+
+					if ($isExeSucc) {
+						Write-Verbose "succeeded to activate License Code." 
+					} else {
+						$retMsg = $ret | Out-String
+						$errMsg = "Failed to activate License Code because " + $retMsg
+						Write-Verbose  $errMsg              
+						throw $errMsg
+					}
+
+					Write-Verbose "Validating License"               
+ 	                $ret = & .\pcoip-validate-license.ps1
+					$isExeSucc = $?
+
+					if ($isExeSucc) {
+						Write-Verbose "succeeded to validate License."
+					} else {
+						$retMsg = $ret | Out-String
+						$errMsg = "Failed to validate license because " + $retMsg
+						Write-Verbose  $errMsg              
+						throw $errMsg
+					}
+                }
+               
+				#start service if it is not started
```

Comment:
Looks good - curious that you had to do this service enabling? That happened sometimes that it was disabled?